Source/ThirdParty/ik/CMakeLists.txt

```diff
@@ -32,6 +37,18 @@ endif ()
 option (IK_MEMORY_DEBUGGING "Global switch for memory options. Keep track of the number of allocations and de-allocations and prints a report when the program shuts down" ${DEFAULT_MEMORY_DEBUGGING})
 cmake_dependent_option (IK_MEMORY_BACKTRACE "Generate backtraces for every malloc(), making it easy to track down memory leaks" "${DEFAULT_MEMORY_DEBUGGING}" "IK_MEMORY_DEBUGGING AND NOT WEB" FALSE)
 
+# Enable restrict keyword in quaternion and vector operations if not in debug
+set (IK_RESTRICT "")
+if (NOT CMAKE_BUILD_TYPE STREQUAL Debug)
```

Comment:
This construct only works for a single-config CMake-generator like Makefile or Ninja. This condition always fails for multi-config generator, such as Xcode and VS.

The following loop may also be potentially slow as although CMake by itself caches the c source compile test result, the way you have written makes the loop to always take place. Perhaps should read to see the variable has already been cached and only loop once in the initial configuration.